pkg/chunked/storage_linux.go

```diff
@@ -1372,6 +1512,30 @@ func (c *chunkedDiffer) ApplyDiff(dest string, options *archive.TarOptions, diff
 		}
 	}()
 
+	if c.convertToZstdChunked {
+		fileSource, diffID, annotations, err := retrieveWholeFile(dest, c.blobSize, c.stream)
+		if err != nil {
+			return graphdriver.DriverWithDifferOutput{}, err
+		}
+		defer fileSource.Close()
+
+		c.stream = fileSource
+
+		manifest, tarSplit, tocOffset, err := readZstdChunkedManifest(c.stream, c.blobSize, annotations)
+		if err != nil {
+			return graphdriver.DriverWithDifferOutput{}, fmt.Errorf("read zstd:chunked manifest: %w", err)
+		}
+
+		c.fileType = fileTypeZstdChunked
+		c.manifest = manifest
+		c.convertToZstdChunked = false
+		c.skipValidation = true
+		c.tarSplit = tarSplit
+		// since we retrieved the whole file, use the diffID instead of the TOC digest.
+		c.tocDigest = diffID
```

Comment:
I’m rather unhappy with naming fields one way and storing entirely different values in them, especially for security-critical ones.

Please define precisely what this value is (“trustedLayerIdentity”? “layerUncompressedDigest”? Something else?), and name (and document) the field that way.

Reply:
sure, added more documentation on what this field means and renamed it as well